kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -102,6 +108,11 @@ public TransactionRebaseState resolveConflicts(Engine engine) throws ConcurrentW
             actionsIterator.forEachRemaining(actionBatch -> {
                 checkArgument(!actionBatch.isFromCheckpoint());  // no checkpoints should be read
                 ColumnarBatch batch = actionBatch.getColumnarBatch();
+                CommitInfo commitInfo = CommitInfo.fromColumnVector(
+                        batch.getColumnVector(COMMITINFO_ORDINAL),0);
+                if (commitInfo != null) {
+                    winningCommitInfoOpt.set(Optional.of(commitInfo));
```

Comment:
I am a bit concerned about the scenario where the last WinningCommit does not have `CommitInfo` but the one before it does. In that case, we will unintentionally end up using the wrong inCommitTimestamp instead of throwing an exception.

Reply:
Thank you for pointing this out! I've updated the code and added a test case to test the scenario where the last WinningCommit does not have `CommitInfo` but the one before it does.